hera_sim/simulate.py

```diff
@@ -129,44 +129,51 @@ def __init__(
                 getattr(self.data, f"get_{attr}"),
             )
 
-        # Now let's expose a few more things to the Simulator
-        for attr in (
-            "antenna_numbers",
-            "ant_1_array",
-            "ant_2_array",
-            "polarization_array",
-        ):
-            setattr(self, attr, getattr(self.data, attr))
+    @property
+    def antenna_numbers(self):
+        return self.data.antenna_numbers
+
+    @property
+    def ant_1_array(self):
+        return self.data.ant_1_array
+
+    @property
+    def ant_2_array(self):
+        return self.data.ant_2_array
+
+    @property
+    def polarization_array(self):
+        return self.data.polarization_array
 
     @property
     def data_array(self):
         """Array storing the visibilities."""
         return self.data.data_array
 
-    @cached_property
+    @property
     def antpos(self):
         """Mapping between antenna numbers and ENU positions in meters."""
         antpos, ants = self.data.get_ENU_antpos(pick_data_ants=True)
         return dict(zip(ants, antpos))
 
-    @cached_property
+    @property
     def lsts(self):
         """Observed Local Sidereal Times in radians."""
         # This process retrieves the unique LSTs while respecting phase wraps.
         _, unique_inds = np.unique(self.data.time_array, return_index=True)
         return self.data.lst_array[unique_inds]
 
-    @cached_property
+    @property
     def freqs(self):
         """Frequencies in GHz."""
         return np.unique(self.data.freq_array) / 1e9
 
-    @cached_property
+    @property
     def times(self):
         """Simulation times in JD."""
         return np.unique(self.data.time_array)
 
-    @cached_property
+    @property
```

Comment:
What is the reason that this was necessary? Operations like `np.unique` are often non-negligible, which is why I cached them originally.

Reply:
Ah, this was so you could do a `select` on the `UVData` object and not have the wrong properties cached. It wasn't necessary for the features developed in this PR, but it was something I had realized as I was making some tweaks to the `Simulator`.

Reply:
Hmm, I guess this is a thing that could happen. I guess having them as `@property` is better for now as it ensures correctness. But we might want to think harder if it becomes a performance problem later on.

Reply:
Hmm. I don't think we'll see much of a difference from not caching these attributes, since the attributes that are determined via a `np.unique` call are only ever queried once per simulation component that's applied (they're pulled from the `Simulator` at the beginning of the `_iteratively_apply` method, before any loops are entered). We can keep an eye on it though.

If it does end up being a non-negligible slowdown, then I think the thing to do is probably make a `select` method for the `Simulator` class that performs the selection on the underlying `UVData` object and then deletes cached attributes that would've been affected by the selection.

Reply:
Agreed. Well, green light from me!